firmware/mcal/stm32f767/periph/pwm.h

```diff
@@ -60,10 +60,22 @@ class PWMOutput : public shared::periph::PWMOutput {
         return float(pulse) / float(period) * 100.0f;
     }
 
+    void SetFrequency(float frequency) override {
+        frequency_ = std::max(0.000015259022f, frequency);
+        uint32_t autoreload = 1/frequency_;
```

Comment:
This expression assumes that the PWM timer is counting at 1Hz, but in reality it is significantly faster and variable. You need to get the timer frequency to correctly perform this calculation.

Consider writing a private `GetTimerFrequency()` method